smoke-tests/src/test/java/io/opentelemetry/smoketest/LinuxTestContainerManager.java

```diff
@@ -0,0 +1,126 @@
+  private static final Logger collectorLogger = LoggerFactory.getLogger("Collector");
+  private static final Logger backendLogger = LoggerFactory.getLogger("Backend");
+
+  private final Network network = Network.newNetwork();
+  private GenericContainer<?> backend = null;
+  private GenericContainer<?> collector = null;
+  private GenericContainer<?> target = null;
+
+  @Override
+  public void startEnvironment() {
+    backend =
+        new GenericContainer<>(
+                DockerImageName.parse(
+                    "ghcr.io/open-telemetry/java-test-containers:smoke-fake-backend-20210324.684269693"))
+            .withExposedPorts(BACKEND_PORT)
+            .waitingFor(Wait.forHttp("/health").forPort(BACKEND_PORT))
+            .withNetwork(network)
+            .withNetworkAliases(BACKEND_ALIAS)
+            .withLogConsumer(new Slf4jLogConsumer(backendLogger));
+    backend.start();
+
+    collector =
+        new GenericContainer<>(DockerImageName.parse("otel/opentelemetry-collector-dev:latest"))
+            .dependsOn(backend)
+            .withNetwork(network)
+            .withNetworkAliases(COLLECTOR_ALIAS)
+            .withLogConsumer(new Slf4jLogConsumer(collectorLogger))
+            .withCopyFileToContainer(
+                MountableFile.forClasspathResource(COLLECTOR_CONFIG_RESOURCE), "/etc/otel.yaml")
+            .withCommand("--config /etc/otel.yaml");
+    collector.start();
+  }
+
+  @Override
+  public void stopEnvironment() {
+    if (backend != null) {
+      backend.stop();
+      backend = null;
+    }
+
+    if (collector != null) {
+      collector.stop();
+      collector = null;
+    }
+
+    network.close();
+  }
+
+  @Override
+  public int getBackendMappedPort() {
+    return backend.getMappedPort(BACKEND_PORT);
+  }
+
+  @Override
+  public int getTargetMappedPort(int originalPort) {
+    return target.getMappedPort(originalPort);
+  }
+
+  @Override
+  public Consumer<OutputFrame> startTarget(
+      String targetImageName,
+      String agentPath,
+      Map<String, String> extraEnv,
+      Map<String, String> extraResources,
+      TargetWaitStrategy waitStrategy) {
+
+    Consumer<OutputFrame> output = new ToStringConsumer();
+    target =
+        new GenericContainer<>(DockerImageName.parse(targetImageName))
+            .withStartupTimeout(Duration.ofMinutes(5))
+            .withExposedPorts(TARGET_PORT)
+            .withNetwork(network)
+            .withLogConsumer(output)
+            .withLogConsumer(new Slf4jLogConsumer(logger))
+            .withCopyFileToContainer(
+                MountableFile.forHostPath(agentPath), "/" + TARGET_AGENT_FILENAME)
+            .withEnv(getAgentEnvironment())
+            .withEnv(extraEnv);
+
+    extraResources.forEach(
+        (file, path) ->
+            target.withCopyFileToContainer(MountableFile.forClasspathResource(file), path));
+
+    if (waitStrategy != null) {
+      if (waitStrategy instanceof TargetWaitStrategy.Log) {
```

Comment:
Is that okay that `TargetWaitStrategy.Http` is not implemented for linux container tests?

Reply:
It's not actually used at all. I don't know why do we have it in Splunk repo in the first place... I will remove it.